### ISSUE TITLE
Give option to make room for Alliance creatures

### ISSUE THREAD

**Issue by zenseii** [ihhub/fheroes2#5249]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

If you have a creature alliance bonus like Dwarven Alliance in the Good Campaign and you don't have room for dwarves, they will simply let you pass and disappear. This is how OG works but fheroes2 lets you choose and potentially dismiss creatures for neutral armies that want to join:
![image](https://user-images.githubusercontent.com/12501091/162645198-3459e268-d0da-418e-aa24-6b5030f966a2.png)

### Describe the possible solution

I suggest implementing an option in the dialog for alliance creatures. There would be some need for reworking of the logic and order of these dialogs.

Like this maybe:
1. They offer to join and you get to choose yes or no and to look at your army
2a. You deny and the message of letting you pass shows.
2b. You accept and they join.

This way you can avoid having them if you don't want them to begin with.

**Comment by LeHerosInconnu** [comment]
@zenseii.

> I suggest implementing an option in the dialog for alliance creatures. There would be some need for reworking of the logic and order of these dialogs.
> 
> Like this maybe:
> 
> 1. They offer to join and you get to choose yes or no and to look at your army
>    2a. You deny and the message of letting you pass shows.
>    2b. You accept and they join.
> 
> This way you can avoid having them if you don't want them to begin with.

Yes, I think a text like this could be added to the current text:

' "My lord, we would also be honored to join your army."
Do you accept?'

In fheroes2:

![Wandering join fh2 02](https://user-images.githubusercontent.com/43583994/116571725-a00ef000-a90b-11eb-9032-8f99231461ec.png)

Similar discussion: https://github.com/ihhub/fheroes2/discussions/3309.

**Comment by zenseii** [comment]
> Could we use common text for joining here? ![image](https://user-images.githubusercontent.com/55348946/163337701-860016ad-81ef-4204-bff9-a9dc15f18536.png) "The dwarves recognize their allies and gladly `want` to join your forces. "Do you accept?" In a case player refuses there would be nice to show the message, which is already implemented (or something similar, I'm concerning about "hail" here). ![image](https://user-images.githubusercontent.com/55348946/163339100-fc43b282-27fa-4c6e-8162-354683b862d9.png)

I agree with the points above. You could remove the part before the quotation marks maybe, not sure if that would look weird.

**Comment by LeHerosInconnu** [comment]
Hello @Branikolog and @zenseii,

> > Could we use common text for joining here? ![image](https://user-images.githubusercontent.com/55348946/163337701-860016ad-81ef-4204-bff9-a9dc15f18536.png) "The dwarves recognize their allies and gladly `want` to join your forces. "Do you accept?" In a case player refuses there would be nice to show the message, which is already implemented (or something similar, I'm concerning about "hail" here). ![image](https://user-images.githubusercontent.com/55348946/163339100-fc43b282-27fa-4c6e-8162-354683b862d9.png)
> 
> I agree with the points above. You could remove the part before the quotation marks maybe, not sure if that would look weird.

Yes, this can work too. :)
